dlt/transformations/transformation.py

```diff
@@ -0,0 +1,194 @@
+import inspect
+from typing import Callable, Any, Type, Optional, cast, Iterator, List
+
+
+import dlt
+
+from dlt.common.typing import TDataItems
+from dlt.common import logger
+
+from dlt.extract.incremental import TIncrementalConfig
+
+from dlt.transformations.typing import (
+    TTransformationFunParams,
+)
+from dlt.transformations.exceptions import (
+    UnknownColumnTypesException,
+    TransformationInvalidReturnTypeException,
+    IncompatibleDatasetsException,
+)
+from dlt.pipeline.exceptions import PipelineConfigMissing
+from dlt.destinations.dataset import ReadableDBAPIDataset
+from dlt.common.schema.typing import TTableSchemaColumns
+from dlt.extract.hints import make_hints
+from dlt.common.destination.dataset import SupportsReadableRelation
+from dlt.extract import DltResource
+from dlt.transformations.configuration import TransformConfiguration
+from dlt.common.utils import get_callable_name
+from dlt.common.schema.typing import (
+    TWriteDisposition,
+    TColumnNames,
+    TSchemaContract,
+    TTableFormat,
+    TTableReferenceParam,
+)
+from dlt.common.configuration import resolve_configuration
+from dlt.common.configuration.specs import known_sections
+from dlt.extract.exceptions import (
+    CurrentSourceNotAvailable,
+)
+
+
+class DltTransformResource(DltResource):
+    def __init__(self, *args: Any, **kwds: Any) -> None:
+        super().__init__(*args, **kwds)
+
+
+def make_transform_resource(
+    func: Callable[TTransformationFunParams, Any],
+    name: str = None,
+    table_name: str = None,
+    write_disposition: TWriteDisposition = None,
+    columns: TTableSchemaColumns = None,
+    primary_key: TColumnNames = None,
+    merge_key: TColumnNames = None,
+    schema_contract: TSchemaContract = None,
+    table_format: TTableFormat = None,
+    references: TTableReferenceParam = None,
+    selected: bool = True,
+    spec: Type[TransformConfiguration] = None,
+    parallelized: bool = False,
+    incremental: Optional[TIncrementalConfig] = None,
+) -> DltTransformResource:
+    # resolve defaults etc
+
+    resource_name = name if name and not callable(name) else get_callable_name(func)
+
+    # check function type, for generators we assume a regular resource
+    is_regular_resource = inspect.isgeneratorfunction(
+        inspect.unwrap(func)
+    ) or inspect.isgeneratorfunction(func)
```

Comment:
is this `inspect.isgeneratorfunction(func)` still needed?

Reply:
no :)